tests/components/fritzbox/conftest.py

```diff
@@ -10,4 +10,5 @@ def fritz_fixture() -> Mock:
     with patch("homeassistant.components.fritzbox.Fritzhome") as fritz, patch(
         "homeassistant.components.fritzbox.config_flow.Fritzhome"
     ):
+        fritz.return_value.get_prefixed_host.return_value = "http://1.2.3.4"
```

Comment:
Why is this test changed?

Reply:
The test used to return a Mock for this function, which resulted in passing Mock to `configuration_url` which triggered invalid device info, and device not to be created.

Reply:
I have slightly changed the behavior here. In the past, an invalid configuration URL would merely result in removal of the invalid configuration URL. With the updated logic, we will actually mark the whole device info as invalid and not create a device.